### PR TITLE
Avoid synchronously waiting for batch trigger execution

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

**Review by jsquire — APPROVED** [review]

File: sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs [hunk 1]
```diff
@@ -184,6 +183,7 @@ private async Task<HttpResponseMessage> ProcessEventsAsync(JArray events, string
                 };
                 executions.Add(_listeners[functionName].Executor.TryExecuteAsync(triggerData, CancellationToken.None));
```
Comment (jsquire) [inline sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs:184]: Was this ever awaited previously, or just accidentally treated as fire-and-forget?

**Comment by pakrym** [comment]
Gimme context.

File: sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs:184]: It was synchronously awaited here - https://github.com/Azure/azure-sdk-for-net/pull/21813/files#diff-00c85e19e67ad5bd236873871ea7d7e5d5aa677cc635bf5af7d94c3848ee7b82R192

**Comment by JoshLove-msft** [comment]
> Gimme context.

I noticed that we weren't awaiting the executions in the batch case. They would still be synchronously awaited in https://github.com/Azure/azure-sdk-for-net/pull/21813/files#diff-00c85e19e67ad5bd236873871ea7d7e5d5aa677cc635bf5af7d94c3848ee7b82R192

**Review by pakrym — APPROVED** [review]

**Pull request merged**